system/src/system_task.cpp

```diff
@@ -368,12 +368,13 @@ void system_delay_ms_non_threaded(unsigned long ms, bool force_no_background_loo
 
 void system_delay_ms(unsigned long ms, bool force_no_background_loop=false)
 {
-    if (system_thread_get_state(NULL)!=0) {
-        HAL_Delay_Milliseconds(ms);
+    if (system_thread_get_state(NULL) == spark::feature::DISABLED &&
```

Comment:
I dunno if a blurb where 15% of the words contain "thread" counts as clarification :laughing: